docs/development/controllers.md

```diff
@@ -2,6 +2,37 @@ Kubernikus Controllers
 ======================
 The Kubernikus operator contains various independent controllers each one handling a different aspect of the managed kubernetes clusters.
 
+Deorbit Controller
+------------------
+
+The Kubernetes `controller-manager` runs controllers that are responsible for
+creating, updating and deleting of OpenStack resources. It auto-provisions
+Cinder volumes for Persistent Volume Claims, load balancers for services and
+routes for nodes in the cluster.  When a kluster is terminated it is desired
+that these resources are cleaned up automatically. 
+
+The problem is that the `controller-manager` only cleans up, when the
+respective resources are deleted from the cluster.
+
+This is where the `deorbiter` comes into play. It inspects the cluster and
+deletes all PVCs that are backed by Cinder, as well as all services of type
+`LoadBalancer`. Afterwards, it waits until the `controller-manager` has
+completed the clean-up.
+
+Problematic is that Cinder volumes can take a variable time until they are
+deleted. In some cases volumes might be stuck deleting indefintelty. The
+`deorbiter` watches the PVs (note: PV not PVC) in the cluster and waits for all
+of them to disappear. PVs are guarded by a finalizer and are only removed when
+the cleanup completes. The `deorbiter` retries the deletion for a configurable
+amount of time and eventually self-destructs the cluster. This might leave some
+debris in the OpenStack project. 
```

Comment:
can we alert on this situation? i think the whole purpose of deorbiting is defeated when there is a chance of this going unnoticed.

Reply:
Yes, totally. There's a counter being incremented for each self-destruction. We can also see, why it was aborted and how it took. I haven't included the alerts yet as I find that hard to dry-code. Wanted to do that in a dedicated PR.